marimo/_runtime/runtime.py

```diff
@@ -1271,27 +1269,22 @@ def mutate_graph(
                 cell_id=cid,
             )
 
-        # Only broadcast Variables message if definitions changed
-        edges_after = self.graph.children
-        definitions_after = set(self.graph.definitions.keys())
-        if (
-            edges_before != edges_after
-            or definitions_before != definitions_after
-        ):
-            Variables(
-                variables=[
-                    VariableDeclaration(
-                        name=variable,
-                        declared_by=list(declared_by),
-                        used_by=list(
-                            self.graph.get_referring_cells(
-                                variable, language="python"
-                            )
-                        ),
-                    )
-                    for variable, declared_by in self.graph.definitions.items()
-                ]
-            ).broadcast()
+        # Always broadcast Variables message after graph mutation to ensure
```

Comment:
This feels expensive, although I'm certain it probably isn't. We do determine references on a per variable level- so in theory we could capture the more granular cases